lib/src/util/useEmptyComponent.ts

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect, useCallback, RefObject } from 'react';
+import useMutationObserver from '@rooks/use-mutation-observer';
+
+const mutationObserverConfig = { subtree: false, childList: true, attributes: false, characterData: false };
+
+/**
+ * Convenience hook that shows if a component is empty or not.
+ * This custom hook implements the useMutationObserver hook to check if there is a change in
+ * the parent component's children.
+ * @return a boolean if the component is empty.
+ * Note: elements within the parent component that are mapped as emptyChildRef are
+ * effectively ignored.
+ */
+const useEmptyComponent = (
```

Comment:
If it's simple, converting `NxTable.Body` to use this could be considered in-scope for this PR.  If you run into problems though (with unit tests most likely), I'd say that's fine to separate out into a tech debt ticket

Reply:
> If it's simple, converting `NxTable.Body` to use this could be considered in-scope for this PR. If you run into problems though (with unit tests most likely), I'd say that's fine to separate out into a tech debt ticket

NxTableBody now uses this custom hook. Didn't have any problems implementing it. Unit tests passed with flying colors.
ff15cc5c615ff606ea72c514b4f3fda183b9f24b